google/async_resumable_media/_download.py

```diff
@@ -377,7 +373,7 @@ def _process_response(self, response):
             callback=self._make_invalid,
         )
         headers = self._get_headers(response)
-        response_body = self._get_body(response)
+        response_body = await self._get_body(response)
```

Comment:
This is exposing my lack of knowledge in async/await. Forgive me for turning this into a learning exercise, but how is it that self._get_headers is sync here but self.get_body is async? If we need to wait for one, doesn't that mean we should also have to wait for the other?